s3_costs/prepare_response.go

```diff
@@ -0,0 +1,155 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package s3_costs
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/trackit/jsonlog"
+	"gopkg.in/olivere/elastic.v5"
+)
+
+// S3BucketCost represents the detail of the costs for a given bucket
+type S3BucketCost struct {
+	GbMonth       float64
+	StorageCost   float64
+	BandwidthCost float64
+	DataIn        float64
+	DataOut       float64
+	RequestsCost  float64
+	Requests      float64
+}
+
+type bucketsInfo = map[string]*S3BucketCost
+type bucket = map[string]interface{}
+
+// isValidBucket is used to filter errored requests that appears in the billing data
+func isValidBucket(bucketName string) bool {
+	return len(bucketName) > 0 && !strings.HasPrefix(bucketName, "[Error:")
+}
+
+// getBucketInfoByName returns the S3BucketCost associated to the bucketName
+func getBucketInfoByName(buckets bucketsInfo, bucketName string) *S3BucketCost {
+	if val, ok := buckets[bucketName]; ok {
+		return val
+	}
+	buckets[bucketName] = &S3BucketCost{}
+	return buckets[bucketName]
+}
+
+func parseBucketsStorage(buckets bucketsInfo, parsedDocument bucket) {
```

Comment:
This function does not return anything. Meaning that it will modify arguments passed to it. This is discouraged, as it makes reading the code less clear.
Either change this function to return the changes, or write a comment explaining what is modified

This comment also applies for all instances of functions that does not return modifications